### PR TITLE
Add ability to expand "other ways to pay" menu

### PR DESCRIPTION
**Pull request by crookedneighbor**
### Summary

Provides an option to auto-expand other ways to pay menu when using a client token with a customer id. This allows the full range of payment options that are available to be used in Drop-in to be shown, instead of _just_ the saved ones.

### Checklist

- [x] Added a changelog entry

### CODE REVIEW

**Comment by crookedneighbor** [comment]
expand

File: src/dropin-model.js [hunk 1]
```diff
@@ -137,6 +137,10 @@ DropinModel.prototype.confirmPaymentMethodDeletion = function (paymentMethod) {
   this._emit('confirmPaymentMethodDeletion', paymentMethod);
 };
 
+DropinModel.prototype.shouldExpandPaymentOptions = function () {
+  return Boolean(this.merchantConfiguration.expandOtherWaysToPay && this.getPaymentMethods().length > 0);
```
Comment (jackellenberger) [inline src/dropin-model.js]: will there ever be a situation where only cards are supported, so `this.getPaymentMethods.length = 1` and we _don't_ want to expand other ways to pay? i.e. should this be kicked up to `> 1`?

Comment (crookedneighbor) [reply src/dropin-model.js]: Good call. This bit is fine as is, but a check for `this.supportedPaymentOptions.length > 1` does need to be added.

**Comment by vaultwiki** [comment]
dropin.js _setUpDependenciesAndViews needs to create this._mainView before the setup of dataCollector and 3dsecure, or asyncDependenciesReady can fire before this._mainView exists, which causes the expandPaymentOptions call there to fail against an undefined object.

This reveals a deeper problem that the individual setups of dataCollector and 3dsecure may finish before the other one queues, which can lead to asyncDependenciesReady firing while one is about to enter the queue. These should be changed so that the initialization is returned as a callback from the individual _setup functions and handled by a finalization step of _setUpDependenciesAndViews. This could explain the random failures of dropin to initialize that I have experienced in the past.

**Comment by crookedneighbor** [comment]
📕 (I trust y'all to handle this when you get to it) 📕

**Pull request closed without merging**